tests/test_kat.py

```diff
@@ -15,12 +16,13 @@ def test_kem(kem_name):
         [helpers.path_to_executable('kat_kem'), kem_name],
     )
     output = output.replace("\r\n", "\n")
+    h256 = sha256()
+    h256.update(output.encode())
     kats = []
     for filename in os.listdir(os.path.join('tests', 'KATs', 'kem')):
-        if filename.startswith(kem_name + '.') and filename.endswith('.kat'):
+        if filename.startswith(kem_name + '.') and filename.endswith('.sha256'):
             with open(os.path.join('tests', 'KATs', 'kem', filename), 'r') as myfile:
-                kats.append(myfile.read())
-    assert(output in kats)
+                assert(myfile.read() == h256.hexdigest())
```

Comment:
This logic could be problematic. In particular, if the KAT file is not present, the test doesn't fail, because the `assert` is guarded by an `if`.  Contrast that with what was present before, where the `assert` always happens.

Why not just immediately try to open the file with the required filename (if the file is not present and the open fails, that's an error anyways), rather than looping through all files to find the one you want?

Reply:
Agree - logic simplified and corrected in new, dict-based version. Please check again.